lib/plugins.ts

```diff
@@ -5,6 +5,21 @@ export interface ILocalStoragePluginState<S extends IState> {
   __version: number
 }
 
+const migrateState = <S extends IState>(oldState: IState, newState: S): S => {
+  const migratedState = {} as S
+
+  for (const key in newState) {
+    const [oldValue, newValue] = [oldState[key], newState[key]]
+    if (key in oldState && typeof oldValue === typeof newValue) {
```

Comment:
I'm not sure if it's necessary to also check the type of a property.
For example, if the type of `id` is changed from `number` to `string` from version 1 to version 2, even though `id` exists in both old and new states, we probably want to use `string` instead of `number` so it's safer.

Reply:
I think this is ok for now, I will add a note in the README that migrations don't automatically check types yet.

Reply:
Oh, I misread your comment at first. I think this is reasonable, but the only way to be sure is to try it out in a real project!